src/controllers/account/account.ts

```diff
@@ -17,35 +22,75 @@ export const generateResetToken = async (): Promise<ResetToken> => {
   return { selector, verifier }
 }
 
-export const getEmailResetToken = async (input: ResetToken): Promise<string> => {
+export const getEmailAuthToken = async (input: AuthToken): Promise<string> => {
   return Buffer.concat([input.selector, input.verifier]).toString('base64')
 }
 
-export const getVerifierHash = async (input: ResetToken): Promise<Buffer> => {
+export const getVerifierHash = async (input: AuthToken): Promise<Buffer> => {
   const verifierHash = Buffer.alloc(sodium.crypto_generichash_BYTES)
   sodium.crypto_generichash(verifierHash, input.verifier)
   return verifierHash
 }
 
-// public static async storeResetToken (selector: Buffer, verifierHash: Buffer) {
-//   // TODO store selector in DB along with hash(verifier)
-// }
+export const storeAuthToken = async (email: string, role: string, selector: Buffer, verifierHash: Buffer) => {
+  // Store email and selector in dynamo DB instance along with hash(verifier)
+  var params = {
+    Item: {
+      email, 
+      role,
+      selector,
+      verifierHash
+    }, 
+    TableName: TABLE_NAME
+  }
+  dynamo.putItem(params, (err: Error, data: {}) => {
```

Comment:
use `dynamo.putItem(...).promise()` to keep this function `async/await` friendly. can use `await dynamo.putItem(...).promise()` in a try/catch

Reply:
done